drivers/serial/uart_async_to_irq.c

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/drivers/serial/uart_async_to_irq.h>
+#include <string.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(UART_ASYNC_TO_IRQ_LOG_NAME, CONFIG_UART_LOG_LEVEL);
+
+/* Internal state flags. */
+
+/* RX interrupt enabled. */
+#define A2I_RX_IRQ_ENABLED	BIT(0)
+
+/* TX interrupt enabled. */
+#define A2I_TX_IRQ_ENABLED	BIT(1)
+
+/* Error interrupt enabled. */
+#define A2I_ERR_IRQ_ENABLED	BIT(2)
+
+/* Receiver to be kept enabled. */
+#define A2I_RX_ENABLE		BIT(3)
+
+/* TX busy. */
+#define A2I_TX_BUSY		BIT(4)
+
+static struct uart_async_to_irq_data *get_data(const struct device *dev)
+{
+	struct uart_async_to_irq_data **data = dev->data;
+
+	return *data;
+}
+
+static const struct uart_async_to_irq_config *get_config(const struct device *dev)
+{
+	const struct uart_async_to_irq_config **config = dev->config;
```

Comment:
Sorry, I missed that one `const` in my previous comment.
```suggestion
	const struct uart_async_to_irq_config *const *config = dev->config;
```